Simple linking error on MacOS

Hello, and first thank you for this amazing piece of software. 

I know MacOS is not supported (yet). But I also know that some people were able to make it work. 

However, I face this seemingly simple linking error when compiling the xcodeproj on MacOS, and I have not been able to resolve it:

```
Setting up Mono
Building MyProjectEditor...
2017-03-14 02:22:51.229 defaults[679:17007533] 
The domain/default pair of (com.apple.dt.Xcode, IDEBuildOperationMaxNumberOfConcurrentCompileTasks) does not exist
Running command : Engine/Binaries/DotNET/UnrealBuildTool.exe MyProjectEditor Mac Development /Users/vincent/Code/Unreal Projects/MyProject/MyProject.uproject
Compiling with MacOSX SDK 10.12
Parsing headers for MyProjectEditor
  Running UnrealHeaderTool "/Users/vincent/Code/Unreal Projects/MyProject/MyProject.uproject" "/Users/vincent/Code/Unreal Projects/MyProject/Intermediate/Build/Mac/MyProjectEditor/Development/MyProjectEditor.uhtmanifest" -LogCmds="loginit warning, logexit warning, logdatabase error" -Unattended -WarningsAsErrors -installed
Reflection code generated for MyProjectEditor in 4.050003 seconds
Performing 4 actions (8 in parallel)
[1/4] Compile Module.AirSim.cpp
[2/4] Compile AirSim.generated.cpp
[3/4] Link UE4Editor-AirSim.dylib
Undefined symbols for architecture x86_64:
  "common_utils::FileSystem::getUserDocumentsFolder()", referenced from:
      common_utils::FileSystem::getAppDataFolder() in Module.AirSim.cpp.o
  "msr::airlib::MavLinkDroneController::MavLinkDroneController()", referenced from:
      msr::airlib::Px4QuadX::createMavController(std::__1::unique_ptr<msr::airlib::DroneControllerBase, std::__1::default_delete<msr::airlib::DroneControllerBase> >&, msr::airlib::SensorCollection&) in Module.AirSim.cpp.o
  "common_utils::FileSystem::createDirectory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)", referenced from:
      common_utils::FileSystem::getLogFileNamePath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) in Module.AirSim.cpp.o
      common_utils::FileSystem::getAppDataFolder() in Module.AirSim.cpp.o
  "msr::airlib::MavLinkDroneController::initialize(msr::airlib::MavLinkDroneController::ConnectionInfo const&, msr::airlib::SensorCollection const*, bool)", referenced from:
      msr::airlib::Px4QuadX::createMavController(std::__1::unique_ptr<msr::airlib::DroneControllerBase, std::__1::default_delete<msr::airlib::DroneControllerBase> >&, msr::airlib::SensorCollection&) in Module.AirSim.cpp.o
  "msr::airlib::Settings::settings_", referenced from:
      MavMultiRotorConnector::startApiServer() in Module.AirSim.cpp.o
      ASimModeBase::BeginPlay() in Module.AirSim.cpp.o
      msr::airlib::Settings::loadJSonFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in Module.AirSim.cpp.o
      msr::airlib::Px4QuadX::getConnectionInfo() in Module.AirSim.cpp.o
  "msr::airlib::RpcLibServer::RpcLibServer(msr::airlib::DroneControllerCancelable*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned short)", referenced from:
      MavMultiRotorConnector::startApiServer() in Module.AirSim.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[4/4] Link UE4Editor-MyProject.dylib
clang: error: no such file or directory: '/Users/vincent/Code/Unreal Projects/MyProject/Plugins/AirSim/Binaries/Mac/UE4Editor-AirSim.dylib'
ERROR: UBT ERROR: Failed to produce item: /Users/vincent/Code/Unreal Projects/MyProject/Plugins/AirSim/Binaries/Mac/UE4Editor-AirSim.dylib
Total build time: 41.56 seconds
Command /Users/Shared/Epic Games/UE_4.14/Engine/Build/BatchFiles/Mac/Build.sh failed with exit code 5
```

Any pointers on what the solution could be ?

--- a/Unreal/Plugins/AirSim/Source/AirSim.Build.cs
+++ b/Unreal/Plugins/AirSim/Source/AirSim.Build.cs
@@ -97,7 +97,7 @@ public class AirSim : ModuleRules
 
     private bool AddLibDependency(string LibName, string LibPath, string LibFileName, TargetInfo Target, bool IsAddLibInclude)
     {
-        string PlatformString = (Target.Platform == UnrealTargetPlatform.Win64) ? "x64" : "x86";
+        string PlatformString = (Target.Platform == UnrealTargetPlatform.Win64 || Target.Platform == UnrealTargetPlatform.Mac) ? "x64" : "x86";
         string ConfigurationString = (Target.Configuration == UnrealTargetConfiguration.Debug) ? "Debug" : "Release";
         bool isLibrarySupported = false;
 
@@ -107,7 +107,7 @@ public class AirSim : ModuleRules
             isLibrarySupported = true;
 
             PublicAdditionalLibraries.Add(Path.Combine(LibPath, PlatformString, ConfigurationString, LibFileName + ".lib"));
-        } else if (Target.Platform == UnrealTargetPlatform.Linux) {
+        } else if (Target.Platform == UnrealTargetPlatform.Linux || Target.Platform == UnrealTargetPlatform.Mac) {
             PublicAdditionalLibraries.Add(Path.Combine(LibPath, "lib" + LibFileName + ".a"));
         }
 

